client/internal/routemanager/notifier.go

```diff
@@ -0,0 +1,86 @@
+package routemanager
+
+import (
+	"sort"
+	"sync"
+
+	"github.com/netbirdio/netbird/route"
+)
+
+// RouteListener is a callback interface for mobile system
+type RouteListener interface {
+	// OnNewRouteSetting invoke when new route setting has been arrived
+	OnNewRouteSetting()
+}
+
+type notifier struct {
+	initialRouteIDs []string
+	ids             []string
+
+	routeListener    RouteListener
+	routeListenerMux sync.Mutex
+}
+
+func newNotifier() *notifier {
+	return &notifier{}
+}
+
+func (n *notifier) setListener(listener RouteListener) {
+	n.routeListenerMux.Lock()
+	defer n.routeListenerMux.Unlock()
+	n.routeListener = listener
+}
+
+func (n *notifier) setInitialClientRoutes(clientRoutes []*route.Route) {
+	ids := make([]string, 0)
+	for _, r := range clientRoutes {
+		ids = append(ids, r.ID)
+	}
+	sort.Strings(ids)
+	n.initialRouteIDs = ids
```

Comment:
we should use unique network ranges instead of IDs as they are the ones that go with interface creation.